flow/layers/opacity_layer.cc

```diff
@@ -44,35 +49,34 @@ void OpacityLayer::Preroll(PrerollContext* context, const SkMatrix& matrix) {
   context->mutators_stack.PushTransform(
       SkMatrix::Translate(offset_.fX, offset_.fY));
   context->mutators_stack.PushOpacity(alpha_);
+
+  AutoCache cache = AutoCache(layer_raster_cache_item_.get(), context, matrix);
   Layer::AutoPrerollSaveLayerState save =
       Layer::AutoPrerollSaveLayerState::Create(context);
 
   // Collect inheritance information on our children in Preroll so that
   // we can decide whether or not to use a saveLayer in Paint.
   context->subtree_can_inherit_opacity = true;
-
   // ContainerLayer will turn the flag off if any children are
   // incompatible or if they overlap
-  ContainerLayer::Preroll(context, child_matrix);
-
+  ContainerLayer::Preroll(context, child_matrix_);
   // We store the inheritance ability of our children for |Paint|
   set_children_can_accept_opacity(context->subtree_can_inherit_opacity);
 
   // Now we let our parent layers know that we, too, can inherit opacity
   // regardless of what our children are capable of
   context->subtree_can_inherit_opacity = true;
-
   context->mutators_stack.Pop();
   context->mutators_stack.Pop();
 
   set_paint_bounds(paint_bounds().makeOffset(offset_.fX, offset_.fY));
 
   if (!children_can_accept_opacity()) {
+    auto child_matrix = child_matrix_;
 #ifndef SUPPORT_FRACTIONAL_TRANSLATION
-    child_matrix = RasterCache::GetIntegralTransCTM(child_matrix);
+    child_matrix = RasterCacheUtil::GetIntegralTransCTM(child_matrix_);
 #endif
-    TryToPrepareRasterCache(context, this, child_matrix,
-                            RasterCacheLayerStrategy::kLayerChildren);
+    layer_raster_cache_item_->CacheChildren(child_matrix);
```

Comment:
Note that if children can accept opacity we shouldn't do any caching at all. Does AutoCache know this?

Reply:
Fixed this problem add a test case: `OpacityLayerTest::ShouldNotCacheChildren`